lib/connection.js

```diff
@@ -357,11 +396,15 @@ function postgresSocket(options, {
       succeeded = true
       i >= options.host.length && (i = 0)
     },
-    write: async x => {
+    pause: () => socket.pause(),
+    resume: () => socket.resume(),
+    isPaused: () => socket.isPaused(),
+    write: async (x, callback) => {
       buffer = buffer ? concat([buffer, x]) : x
       if (buffer.length >= 1024)
-        await write()
+        await write(callback)
       next === null && (next = setImmediatePromise(write)).catch(error)
+      callback && callback()
```

Comment:
Unhandled promise rejection again if `callback` returns a rejected `Promise`
```suggestion
      return callback && callback()
```